Introduce Unittests

sedrubal opened the pull request:
Replaces #9

TODO:
- [x] We should add codecov or another unit test web app and badges
- [x] We should add more unittests (but this can be done after merging)

sedrubal committed:
Introduce unittests

 - add test for argparse
 - integrate test in setup.py
 - update docu
 - Renamed cli script, fixes #7

sedrubal committed:
Rename argparse test, remove starting tests as script

sedrubal committed:
Merge branch 'develop' into unittests

- So we can test, if identifier is readonly

sedrubal committed:
Merge branch 'develop' into unittests

Because we can only test the print_tree function if it returns a str

sedrubal committed:
Add ImageLayer unittests

sedrubal committed:
Merge branch 'develop' into unittests

sedrubal committed:
Add tests for size prop and _convert_size, improve _convert_size

sedrubal committed:
Remove print_function import as no print is used

nightvisi0n committed:
fix error in unittests link

Signed-off-by: julian <dev@jneureuther.de>

nightvisi0n commented:
> We should add codecov or another unit test web app and badges

i will implement that before merging

nightvisi0n committed:
add travis test config

Signed-off-by: julian <dev@jneureuther.de>

nightvisi0n committed:
import modules directly in files containing a test for it

the previous implementation was broken when used in combination with coverage

Signed-off-by: julian <dev@jneureuther.de>

nightvisi0n committed:
add script which executes all unit tests

this is used when executing code coverage checks in combination with coverage.py

Signed-off-by: julian <dev@jneureuther.de>

nightvisi0n committed:
fix error in path insertion

the tests are normally (e.g. when executed by travis-ci) run from the projects root directory, so we need to add the current directory ('.') to path, not the parent directory ('..').

Signed-off-by: julian <dev@jneureuther.de>

nightvisi0n committed:
install docktree and all dependencies during travis build

Signed-off-by: julian <dev@jneureuther.de>

nightvisi0n committed:
add code coverage and build status badges

Signed-off-by: julian <dev@jneureuther.de>

nightvisi0n commented:
> We should add more unittests (but this can be done after merging)

I would also suggest to do this in another pull request.

Everything else: LGTM, @sedrubal can you please review my additions? :smile:

sedrubal committed:
Test also other python versions

sedrubal committed:
Don't support python 3.2

sedrubal committed:
Remove context.py (frickel frickel)

sedrubal committed:
Automagically load tests/test_*.py for coverage

hacky
